### PR TITLE
lib/model: Handle type changes when pulling (ref #4505 #4506 #4507)

### PR DESCRIPTION
**Pull request by imsodin**
This is a variant of #4507 that should fix the problems raised there by Audrius.

### CODE REVIEW

**Comment by calmh** [comment]
With this we only check permissions on directories, not files?

**Comment by imsodin** [comment]
The files case also tests permissions by `fallthrough` to the dir case.

File: lib/model/rwfolder.go [hunk 1]
```diff
@@ -1391,26 +1391,43 @@ func (f *sendReceiveFolder) performFinish(state *sharedPullerState) error {
...
-			// has changed, rescan.
-			case !state.hasCurFile || !stat.ModTime().Equal(state.curFile.ModTime()) || stat.Size() != state.curFile.Size:
-				fallthrough
-
-			// Permissions have changed, means the file has changed, rescan.
-			case !f.ignorePermissions(state.curFile) && state.curFile.HasPermissionBits() && !scanner.PermsEqual(state.curFile.Permissions, curMode):
+		changed := false
+		switch {
+		case !state.hasCurFile:
+			// The file appeared from nowhere
```
Comment (AudriusButkevicius) [inline lib/model/rwfolder.go]: This is changed too?

Comment (imsodin) [reply lib/model/rwfolder.go]: Yes of course, thanks for noticing.

Comment (AudriusButkevicius) [reply lib/model/rwfolder.go]: Perhaps log too?

Comment (imsodin) [reply lib/model/rwfolder.go]: done

**Comment by AudriusButkevicius** [comment]
@st-review lgtm

**Comment by st-review** [comment]
@AudriusButkevicius: Noted! Need another LGTM or explicit merge command.

**Comment by calmh** [comment]
@st-review lgtm

**Comment by st-review** [comment]
:ok_hand: Merged as fa12a18190dc4db90b6a8ec7860ff29f849567a1. Thanks, @imsodin!

**Pull request closed without merging**